### PR TITLE
GridDataOutput: some cleanup

### PR DESCRIPTION
**Pull request by akva2**
- add translation unit
- avoid stringstream member
- add const correctness
- fix some formatting

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/utils/GridDataOutput_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+    return i;
+}
+
+template <class GridView, unsigned int partitions>
+template <typename T>
+long SimMeshDataAccessor<GridView,partitions>::
+writeGridPoints_AOS(T* xyz_inout, long max_size) const
+{
+    if (max_size < nvertices_ * 3) {
+        assert(max_size >= nvertices_ * 3);
```
Comment (bska) [inline opm/simulators/utils/GridDataOutput_impl.hpp:160]: Tiny nit: We should `#include <cassert>` to bring in a declaration of the `assert` macro.

File: opm/simulators/utils/GridDataOutput_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+    }
+    return false;
+}
+
+template <class GridView, unsigned int partitions>
+void SimMeshDataAccessor<GridView,partitions>::countEntities()
+{
+    // We include all the vertices for this ranks partition
+    const auto& vert_partition = vertices(gridView_, Dune::Partitions::all);
+    nvertices_ = std::distance(vert_partition.begin(), vert_partition.end());
```
Comment (bska) [inline opm/simulators/utils/GridDataOutput_impl.hpp:59]: I think we need to `#include <algorithm>` to bring in a declaration of `std::distance()`.

**Review by bska — APPROVED** [review]
Looks good to me apart from two missing `#include` statements.

File: opm/simulators/utils/GridDataOutput_impl.hpp (hunk 2 above)
Comment (akva2) [reply opm/simulators/utils/GridDataOutput_impl.hpp:61]: actually it's defined in ```<iterator>``` - included that one.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**